pkg/diff/diff_test.go

```diff
@@ -149,6 +149,22 @@ func TestDiff(t *testing.T) {
 	}
 }
 
+func TestDiff_KnownTypeInvalidValue(t *testing.T) {
+	leftDep := newDeployment()
+	leftUn := mustToUnstructured(leftDep)
+	if !assert.NoError(t, unstructured.SetNestedField(leftUn.Object, "badValue", "spec", "revisionHistoryLimit")) {
+		return
+	}
+
+	diffRes := diff(t, leftUn, leftUn, GetDefaultDiffOptions())
```

Comment:
This is intentional `leftUn ` is used as input parameter of both `config` and `live`?

Reply:
Yes, this test ensures that diff returns false (not different) for the equal objects even if object has "invalid" fields.

Reply:
implemented an additional use case to ensure that diff correctly handles different resources with invalid fields as well.